### PR TITLE
Adding System.Net.Mail to the uap and uapaot closures

### PR DESCRIPTION
**Pull request by joperezr**
cc: @danmosemsft 

related issue: #16872

This solves just part of the issue since we still need to fix System.Net.HttpListener, but I'll do that in a separate PR.

### CODE REVIEW

File: src/System.Security.AccessControl/src/System.Security.AccessControl.csproj [hunk 1]
```diff
@@ -9,16 +9,18 @@
...
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp-Windows_NT-Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netfx-Windows_NT-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netfx-Windows_NT-Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Windows_NT-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Windows_NT-Release|AnyCPU'" />
   <ItemGroup Condition="'$(TargetGroup)'=='netfx'">
     <Reference Include="mscorlib" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetGroup)'=='netcoreapp' AND '$(TargetsWindows)'=='true'">
+  <ItemGroup Condition="('$(TargetGroup)'=='netcoreapp' OR '$(TargetGroup)'=='uap') AND '$(TargetsWindows)'=='true'">
```
Comment (danmosemsft) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: what about uapaot?

File: src/System.Security.Permissions/src/System.Security.Permissions.csproj [hunk 2]
```diff
@@ -9,11 +9,13 @@
...
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='netcoreapp-Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='netcoreapp-Release|AnyCPU'" />
-  <ItemGroup Condition="'$(TargetGroup)' == 'netcoreapp'">
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp-Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netfx-Windows_NT-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netfx-Windows_NT-Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Windows_NT-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Windows_NT-Release|AnyCPU'" />
+  <ItemGroup Condition="'$(TargetGroup)' == 'netcoreapp' or '$(TargetGroup)' == 'uap'">
```
Comment (danmosemsft) [inline src/System.Security.Permissions/src/System.Security.Permissions.csproj]: uapaot?

File: src/System.Security.AccessControl/src/System.Security.AccessControl.csproj (hunk 1 above)
Comment (jkotas) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: Not needed. We should be adding uapaot conditions only where they are actually needed.

Comment (danmosemsft) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: Right that was before our discussion.

File: src/System.Security.Permissions/src/System.Security.Permissions.csproj (hunk 2 above)
Comment (danmosemsft) [inline src/System.Security.Permissions/src/System.Security.Permissions.csproj]: ignore

**Pull request merged**